flambe/field/text.py

```diff
@@ -65,6 +65,46 @@ def get_embeddings(
     return model
 
 
+class EmbeddingsInformation(NamedTuple):
+    """
+    Information about an embedding model.
+
+    Parameters
+    ----------
+    embeddings : str
+        Path to pretrained embeddings or the embedding name
+        in case format is gensim.
+    embeddings_format : str, optional
+        The format of the input embeddings, should be one of:
+        'glove', 'word2vec', 'fasttext' or 'gensim'. The latter can
+        be used to download embeddings hosted on gensim on the fly.
+        See https://github.com/RaRe-Technologies/gensim-data
+        for the list of available embedding aliases.
+    embeddings_binary : bool, optional
+        Whether the input embeddings are provided in binary format,
+        by default False
+    setup_all_embeddings: bool
+        Controls if all words from the optional provided
+        embeddings will be added to the vocabulary and to the
+        embedding matrix. Defaults to False.
+    unk_init_all : bool, optional
+        If True, every token not provided in the input embeddings is
+        given a random embedding from a normal distribution.
+        Otherwise, all of them map to the '<unk>' token.
+    drop_unknown: bool
+        Whether to drop tokens that don't have embeddings
+        associated. Defaults to True.
+        Important: this flag will only work when using embeddings.
+
+    """
+    embeddings: str
+    embeddings_format: str = 'gensim'
+    embeddings_binary: bool = False
+    setup_all_embeddings: bool = False
```

Comment:
I don't like this parameter name, it's not clear that it has to do with using the embeddings as source of the vocabulary

Reply:
`embeddings_as_vocab_source`? Better ideas?

Reply:
`use_all_from_vectored_keys`?

Reply:
Lets do `build_vocab_from_embeddings`.

Already discussed with the team, this is a harmless backwards incompatible change.

Reply:
Actually, no backwards incompatibility as we keep `setup_all_embeddings` in the init